python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -146,6 +152,8 @@ def action_triggered():
             # the display name of the command
             menu_name = None
             button_name = title
+            found_collapse_match = False
```

Comment:
it would be great with some comments here (with examples) explaining the collapsing and the grouping, how they relate and what takes precedence. This code was already kind of hard to read and now we just added more complexity, so a bit of comments to help out future code readers i think would be really helpful!